File: android/src/main/java/expo/modules/nativemenus/NativeMenusModule.kt
package expo.modules.nativemenus

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeMenusModule : Module() {

    override fun definition() = ModuleDefinition {

        Name("NativeMenus")


        View(NativeMenusView::class) {
            Events("onPressEvent")

            Prop("menuItems") { view: NativeMenusView, prop: ArrayList<MenuItemProps> ->
                view.setData(prop)
            }
        }
    }
}

File: ios/NativeMenusModule.swift
import ExpoModulesCore

public class NativeMenusModule: Module {

  public func definition() -> ModuleDefinition {

    Name("NativeMenus")

    View(NativeMenusView.self) {
      // Defines a setter for the `name` prop.
        Prop("menuItems") { (view: NativeMenusView, prop: [MenuItemProps]) in
            view.viewModel.menuItems = prop
      }
    }
  }
}
